src/psij/executors/batch_test.py

```diff
@@ -0,0 +1,89 @@
+import sys
+from distutils.version import StrictVersion
+from pathlib import Path
+from typing import Optional, Collection, List, Dict, TextIO
+
+from psij import Job, JobStatus, JobState
+from psij.executors.batch.batch_scheduler_executor import BatchSchedulerExecutor, \
+    BatchSchedulerExecutorConfig
+from psij.executors.batch.script_generator import TemplatedScriptGenerator
+
+
+QSUB_PATH = str(Path(__file__).parent / 'batch' / 'test' / 'qsub')
+QSTAT_PATH = str(Path(__file__).parent / 'batch' / 'test' / 'qstat')
+QDEL_PATH = str(Path(__file__).parent / 'batch' / 'test' / 'qdel')
+
+
+class _TestExecutorConfig(BatchSchedulerExecutorConfig):
+    """A configuration class for the test batch executor."""
+
+    def __init__(self, launcher_log_file: Optional[Path] = None,
+                 work_directory: Optional[Path] = None, queue_polling_interval: int = 1,
+                 queue_polling_error_threshold: int = 2, keep_files: bool = False):
+        """Initializes a `TestExecutorConfig`.
+
+        This constructor overrides the constructor for :class:`~BatchSchedulerExecutorConfig` to
+        change the `queue_polling_interval` default to 5 seconds.
+        """
+        super().__init__(launcher_log_file, work_directory, queue_polling_interval,
+                         queue_polling_error_threshold, keep_files)
+
+
+class _TestJobExecutor(BatchSchedulerExecutor):
+    _NAME_ = 'batch-test'
+    _VERSION_ = StrictVersion('0.0.1')
+
+    _STATE_MAP = {
+        'F': JobState.FAILED,
+        'X': JobState.CANCELED,
+        'C': JobState.COMPLETED,
+        'Q': JobState.QUEUED,
+        'R': JobState.ACTIVE,
+    }
+
+    def __init__(self, url: Optional[str] = None, config: Optional[_TestExecutorConfig] = None):
+        if not config:
+            config = _TestExecutorConfig()
+        super().__init__(config=config)
+        self.generator = TemplatedScriptGenerator(config, Path(__file__).parent / 'batch' / 'test'
+                                                  / 'test.mustache')
+
+    def generate_submit_script(self, job: Job, context: Dict[str, object],
+                               submit_file: TextIO) -> None:
+        self.generator.generate_submit_script(job, context, submit_file)
+
+    def get_submit_command(self, job: Job, submit_file_path: Path) -> List[str]:
+        return [sys.executable, QSUB_PATH, str(submit_file_path.absolute())]
+
+    def get_cancel_command(self, job: Job) -> List[str]:
+        assert job.native_id
+        return [sys.executable, QDEL_PATH, job.native_id]
+
+    def get_status_command(self, jobs: Collection[Job]) -> List[str]:
+        ids = self.join_native_ids(jobs, ',')
+
+        # we're not really using job arrays, so this is equivalent to the job ID. However, if
+        # we were to use arrays, this would return one ID for the entire array rather than
+        # listing each element of the array independently
```

Comment:
Is this comment an errant copy-paste leftover from the slurm executor?

Reply:
Indeed. Thanks.

Reply:
I removed it. Thanks again.